internal/databases/mysql/stream_fetch_handler.go

```diff
@@ -84,55 +126,86 @@ func downloadAndDecompressStream(folder storage.Folder, fileName string) error {
 	return internal.NewArchiveNonExistenceError(fmt.Sprintf("Archive '%s' does not exist.\n", fileName))
 }
 
-func fetchBinlogs(folder storage.Folder, backupStartUploadTime time.Time, binlogsAreDone chan error) {
+func getNecessaryBinlog(folder storage.Folder, backupStartUploadTime time.Time) (fetchedLogs []storage.Object, err error) {
 	binlogFolder := folder.GetSubFolder(BinlogPath)
 	endTS, dstFolder, err := GetBinlogConfigs()
+
 	if err != nil {
-		binlogsAreDone <- err
-		return
+		return nil, err
 	}
 	objects, _, err := binlogFolder.ListFolder()
+
 	if err != nil {
-		binlogsAreDone <- nil
-		return
+		return nil, err
 	}
-	var fetchedLogs []storage.Object
+
+	tracelog.InfoLogger.Println("Times ", backupStartUploadTime, endTS)
 
 	for _, object := range objects {
-		tracelog.InfoLogger.Println("Consider binlog ", object.GetName(), object.GetLastModified().Format(time.RFC3339))
 
 		binlogName := ExtractBinlogName(object, folder)
+		fileName := path.Join(dstFolder, binlogName)
+		tracelog.InfoLogger.Println("Download", binlogName, "to", fileName)
+		err := internal.DownloadWALFileTo(binlogFolder, binlogName, fileName)
```

Comment:
We need to download object only if it's upload date >= backupStartUploadTime
Check it before downloading.